docs/fsharp-cheatsheet.md

```diff
@@ -740,6 +743,112 @@ Another way of implementing interfaces is to use *object expressions*.
 
 *Partial active patterns* share the syntax of parameterized patterns but their active recognizers accept only one argument.
 
+<div id="asynchronous-programming"></div>
+
+## Asynchronous Programming
+
+F# asynchronous programming is centered around two core concepts: async computations and tasks.
+
+    async {
+        do! Async.Sleep (waitInSeconds * 1000)
+        let! asyncResult = asyncComputation
+        use! disposableResult = iDisposableAsyncComputation
+    }
+
+### Async vs Tasks
+
+An async computation is a unit of work, and a task is a promise of a result. A subtle but important distinction.
```

Comment:
a unit of work is a term that makes most people think of POSA and/or entity framework
I think hot vs cold tends to work as a way of explaining things.
I tend to explain it as:
- Async is a builder pattern that lets you start N copies of a thing, but nothing is started until you ask
- Task is like a Promise - it's a handle for the outcome of some work (which might still be in progress, or not)

This way of expressing it does not speak to me. But then I'm not the audience ;)

Reply:
I think I'd be showing Asyc.Parallel vs Async.Sequential, and maybe some stuff about how you do equivalent things to convey that they are their own separate animals, and it's not as simple as "converting" anything

Reply:
"An async computation is a specification of work that isn't started until requested; and a task is a promise of a result from an operation, which might be waiting, running, or completed" ?